### PR TITLE
[WIP] Add sqlx support

### PR DESCRIPTION
**Pull request by godwhoa**
This PR is still work-in-progress. Currently running into an issue with the analysistest suite.
```
➜  sqlargs git:(sqlx2) go test
--- FAIL: TestSqlx (0.04s)
    analysistest.go:103: loading [sqlx]: go [list -e -json -compiled -test=true -export=false -deps=true -find=false -- sqlx]: exit status 1: go build github.com/jmoiron/sqlx: no Go files in 
FAIL
exit status 1
FAIL	github.com/agnivade/sqlargs	1.772s
```
One solution I have found is to vendor sqlx into `testdata/` but that probably isn't ideal. So, looking for suggestions

### CODE REVIEW

File: sqlargs_test.go [hunk 1]
```diff
@@ -1,18 +1,53 @@
-package sqlargs_test
+package sqlargs
```
Comment (agnivade) [inline sqlargs_test.go:1]: Please keep this `sqlargs_test`.

**Review by agnivade — CHANGES_REQUESTED** [review]

**Comment by agnivade** [comment]
Thanks ! 

But unfortunately, this will accept patterns like (*sql.DB).MustExecContext or (*sql.DB).QueryxContext which we don't want. I was thinking of a more generalized approach to this by moving out the selector expressions to be checked in a separate struct and making the check logic generic.

Also, for sqlx support, you can just run a `go get <>` in the .travis.yml file to import the dep.

**Comment by agnivade** [comment]
I will keep your PR on-hold for some time till I can generalize the selector expression check. Then we can merge this. Thanks for the vendor check btw; it hadn't occurred to me.

File: sqlargs_test.go (hunk 1 above)
Comment (godwhoa) [reply sqlargs_test.go:1]: Made it part of `sqlargs` package for testing `stripVendor`.

**Comment by godwhoa** [comment]
>I will keep your PR on-hold for some time till I can generalize the selector expression check.

Sure :+1:

File: sqlargs_test.go (hunk 1 above)
Comment (agnivade) [reply sqlargs_test.go:1]: Ah, alright then.

**Comment by agnivade** [comment]
Hey @godwhoa - Now latest master should be ready to add sqlx support. Could you please get the latest and apply your changes ?

EDIT: You just need to add a separate package object like here https://github.com/agnivade/sqlargs/blob/master/sqlargs.go#L34

**Comment by godwhoa** [comment]
@agnivade Sure

**Comment by agnivade** [comment]
Thanks @godwhoa. I just have one more small request. Since you merged master into your branch, now I cannot rebase this branch into master. 

To keep the master branch clean, I would request you to have just 2 commits in this branch, one which adds sqlx support, the other adding the stripvendor check.

Or if you want me to squash the commits, I am fine with it too. But keeping 2 separate commits would be ideal. Thanks.

**Comment by godwhoa** [comment]
Got it, my git skills are limited but I'll try.

Edit: Okay I give up. Lets go with the 3rd option of squashing the commits.

**Review by agnivade — APPROVED** [review]

**Pull request merged**